File: swt4ruby/src/x86_64-linux/org/eclipse/swt/internal/mozilla/nsISecurityCheckedComponent.java
package org.eclipse.swt.internal.mozilla;


public class nsISecurityCheckedComponent extends nsISupports {

	static final int LAST_METHOD_ID = nsISupports.LAST_METHOD_ID + 4;

	public static final String NS_ISECURITYCHECKEDCOMPONENT_IID_STR =
		"0dad9e8c-a12d-4dcb-9a6f-7d09839356e1";

	public static final nsID NS_ISECURITYCHECKEDCOMPONENT_IID =
		new nsID(NS_ISECURITYCHECKEDCOMPONENT_IID_STR);

	public nsISecurityCheckedComponent(long /*int*/ address) {
		super(address);
	}

	public int CanCreateWrapper(long /*int*/ iid, long /*int*/[] _retval) {
		return XPCOM.VtblCall(nsISupports.LAST_METHOD_ID + 1, getAddress(), iid, _retval);
	}

	public int CanCallMethod(long /*int*/ iid, char[] methodName, long /*int*/[] _retval) {
		return XPCOM.VtblCall(nsISupports.LAST_METHOD_ID + 2, getAddress(), iid, methodName, _retval);
	}

	public int CanGetProperty(long /*int*/ iid, char[] propertyName, long /*int*/[] _retval) {
		return XPCOM.VtblCall(nsISupports.LAST_METHOD_ID + 3, getAddress(), iid, propertyName, _retval);
	}

	public int CanSetProperty(long /*int*/ iid, char[] propertyName, long /*int*/[] _retval) {
		return XPCOM.VtblCall(nsISupports.LAST_METHOD_ID + 4, getAddress(), iid, propertyName, _retval);
	}
}

File: swt4ruby/lib/x86_64-linux/org/eclipse/swt/internal/mozilla/NsISecurityCheckedComponent.rb
require "rjava"

module Org::Eclipse::Swt::Internal::Mozilla
  module NsISecurityCheckedComponentImports #:nodoc:
    class_module.module_eval {
      include ::Java::Lang
      include ::Org::Eclipse::Swt::Internal::Mozilla
    }
  end
  
  class NsISecurityCheckedComponent < NsISupports
    include_class_members NsISecurityCheckedComponentImports
    
    class_module.module_eval {
      const_set_lazy(:LAST_METHOD_ID) { NsISupports::LAST_METHOD_ID + 4 }
      const_attr_reader  :LAST_METHOD_ID
      
      const_set_lazy(:NS_ISECURITYCHECKEDCOMPONENT_IID_STR) { "0dad9e8c-a12d-4dcb-9a6f-7d09839356e1" }
      const_attr_reader  :NS_ISECURITYCHECKEDCOMPONENT_IID_STR
      
      const_set_lazy(:NS_ISECURITYCHECKEDCOMPONENT_IID) { NsID.new(NS_ISECURITYCHECKEDCOMPONENT_IID_STR) }
      const_attr_reader  :NS_ISECURITYCHECKEDCOMPONENT_IID
    }
    
    typesig { [::Java::Long] }
    # int
    def initialize(address)
      super(address)
    end
    
    typesig { [::Java::Long, Array.typed(::Java::Long)] }
    # int
    # int
    def _can_create_wrapper(iid, _retval)
      return XPCOM._vtbl_call(NsISupports::LAST_METHOD_ID + 1, get_address, iid, _retval)
    end
    
    typesig { [::Java::Long, Array.typed(::Java::Char), Array.typed(::Java::Long)] }
    # int
    # int
    def _can_call_method(iid, method_name, _retval)
      return XPCOM._vtbl_call(NsISupports::LAST_METHOD_ID + 2, get_address, iid, method_name, _retval)
    end
    
    typesig { [::Java::Long, Array.typed(::Java::Char), Array.typed(::Java::Long)] }
    # int
    # int
    def _can_get_property(iid, property_name, _retval)
      return XPCOM._vtbl_call(NsISupports::LAST_METHOD_ID + 3, get_address, iid, property_name, _retval)
    end
    
    typesig { [::Java::Long, Array.typed(::Java::Char), Array.typed(::Java::Long)] }
    # int
    # int
    def _can_set_property(iid, property_name, _retval)
      return XPCOM._vtbl_call(NsISupports::LAST_METHOD_ID + 4, get_address, iid, property_name, _retval)
    end
    
    private
    alias_method :initialize_ns_isecurity_checked_component, :initialize
  end
  
end
